sklearn/ensemble/partial_dependence.py

```diff
@@ -306,8 +306,8 @@ def convert_feature(fx):
                 l.append(feature_names[i])
             names.append(l)
     except IndexError:
-        raise ValueError('features[i] must be in [0, n_features) '
-                         'but was %d' % i)
+        raise ValueError('features[i] must be in [0, len(feature_names)={0} ) '
+                         'but was {1}'.format(len(feature_names), i))
```

Comment:
Why not ``features contains {} which is too large for feature_names``? or something like that?
Maybe ``"All entries of features must be <len(feature_names)={}, got {}".format(len(feature_names), features[i])"``